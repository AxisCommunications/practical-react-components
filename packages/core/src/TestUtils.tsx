import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import { PracticalProvider } from './Practical'

/*
  Only used for Jest Snapshot Testing
*/

export const TP = ({
  children,
}: {
  readonly children: React.ReactElement
}): JSX.Element => <PracticalProvider>{children}</PracticalProvider>

/**
 * Mock reference to practical-root
 */
const createNodeMock = (element: React.ReactElement) => {
  if (element.props.id !== 'practical-root') {
    return null
  }

  return {}
}

export const TestRender = (jsx: JSX.Element) => {
  return renderer.create(<TP>{jsx}</TP>, { createNodeMock }).toJSON()
}

export const TestText = (): JSX.Element => <span>Test</span>

/* eslint-disable-next-line react/display-name */
export const TestTextWithForwardRef = React.forwardRef<HTMLElement>(
  ({ ...props }, ref) => (
    <span {...props} ref={ref}>
      Test
    </span>
  )
)
