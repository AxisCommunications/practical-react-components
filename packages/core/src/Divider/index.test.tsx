import React from 'react'
import 'jest-styled-components'

import { Divider } from '.'
import { TestRender } from '../TestUtils'

describe('Dividers', () => {
  test('Divider variant', () => {
    const tree1 = TestRender(<Divider />)
    expect(tree1).toMatchSnapshot()

    const tree2 = TestRender(<Divider variant="full" />)
    expect(tree2).toMatchSnapshot()

    const tree3 = TestRender(<Divider variant="middle" />)
    expect(tree3).toMatchSnapshot()

    const tree4 = TestRender(<Divider variant="inset" />)
    expect(tree4).toMatchSnapshot()

    const tree5 = TestRender(<Divider variant="inset" insetSize={100} />)
    expect(tree5).toMatchSnapshot()
  })
})
