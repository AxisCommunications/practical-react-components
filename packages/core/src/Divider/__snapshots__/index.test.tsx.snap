// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dividers Divider variant 1`] = `
.c2 {
  height: 1px;
  background-color: rgb(230,230,230);
}

.c4 {
  position: fixed;
  top: 0;
  width: 360px;
  border-radius: 4px;
  margin: 8px 16px;
  right: 0;
}

.c0 {
  color: rgb(41,41,41);
}

.c1 {
  position: absolute;
  z-index: 0;
}

.c3 {
  position: absolute;
  z-index: 1;
}

<div
  className="c0"
  id="practical-root"
>
  <div>
    <div
      className="c1"
      id="layer-1"
    >
      <div
        className="c2"
      />
    </div>
  </div>
  <div>
    <div
      className="c3"
      id="layer-2"
    >
      <div
        className="c4"
      />
    </div>
  </div>
</div>
`;

exports[`Dividers Divider variant 2`] = `
.c2 {
  height: 1px;
  background-color: rgb(230,230,230);
}

.c4 {
  position: fixed;
  top: 0;
  width: 360px;
  border-radius: 4px;
  margin: 8px 16px;
  right: 0;
}

.c0 {
  color: rgb(41,41,41);
}

.c1 {
  position: absolute;
  z-index: 0;
}

.c3 {
  position: absolute;
  z-index: 1;
}

<div
  className="c0"
  id="practical-root"
>
  <div>
    <div
      className="c1"
      id="layer-3"
    >
      <div
        className="c2"
      />
    </div>
  </div>
  <div>
    <div
      className="c3"
      id="layer-4"
    >
      <div
        className="c4"
      />
    </div>
  </div>
</div>
`;

exports[`Dividers Divider variant 3`] = `
.c2 {
  height: 1px;
  background-color: rgb(230,230,230);
  margin: 0 32px;
}

.c4 {
  position: fixed;
  top: 0;
  width: 360px;
  border-radius: 4px;
  margin: 8px 16px;
  right: 0;
}

.c0 {
  color: rgb(41,41,41);
}

.c1 {
  position: absolute;
  z-index: 0;
}

.c3 {
  position: absolute;
  z-index: 1;
}

<div
  className="c0"
  id="practical-root"
>
  <div>
    <div
      className="c1"
      id="layer-5"
    >
      <div
        className="c2"
      />
    </div>
  </div>
  <div>
    <div
      className="c3"
      id="layer-6"
    >
      <div
        className="c4"
      />
    </div>
  </div>
</div>
`;

exports[`Dividers Divider variant 4`] = `
.c2 {
  height: 1px;
  background-color: rgb(230,230,230);
  margin-left: 64px;
}

.c4 {
  position: fixed;
  top: 0;
  width: 360px;
  border-radius: 4px;
  margin: 8px 16px;
  right: 0;
}

.c0 {
  color: rgb(41,41,41);
}

.c1 {
  position: absolute;
  z-index: 0;
}

.c3 {
  position: absolute;
  z-index: 1;
}

<div
  className="c0"
  id="practical-root"
>
  <div>
    <div
      className="c1"
      id="layer-7"
    >
      <div
        className="c2"
      />
    </div>
  </div>
  <div>
    <div
      className="c3"
      id="layer-8"
    >
      <div
        className="c4"
      />
    </div>
  </div>
</div>
`;

exports[`Dividers Divider variant 5`] = `
.c2 {
  height: 1px;
  background-color: rgb(230,230,230);
  margin-left: 100px;
}

.c4 {
  position: fixed;
  top: 0;
  width: 360px;
  border-radius: 4px;
  margin: 8px 16px;
  right: 0;
}

.c0 {
  color: rgb(41,41,41);
}

.c1 {
  position: absolute;
  z-index: 0;
}

.c3 {
  position: absolute;
  z-index: 1;
}

<div
  className="c0"
  id="practical-root"
>
  <div>
    <div
      className="c1"
      id="layer-9"
    >
      <div
        className="c2"
      />
    </div>
  </div>
  <div>
    <div
      className="c3"
      id="layer-10"
    >
      <div
        className="c4"
      />
    </div>
  </div>
</div>
`;
